File: articles/probability-calibration-and-decision-confidence/python/run_all_calibration_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Probability Calibration and Decision Confidence."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "brier_score_calculator.py",
    "log_loss_calculator.py",
    "reliability_table_builder.py",
    "expected_calibration_error.py",
    "confidence_bias_diagnostics.py",
    "decision_threshold_calibration.py",
    "base_rate_reference_class_checks.py",
    "decision_record_exporter.py",
    "probability_calibration_decision_confidence_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python calibration workflows completed.")


if __name__ == "__main__":
    main()

File: articles/probability-calibration-and-decision-confidence/r/run_all_calibration_workflows.R
# run_all_calibration_workflows.R
# Run all R workflows for Probability Calibration and Decision Confidence.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "probability_calibration_decision_confidence_workflow.R",
  "brier_score_profiles.R",
  "reliability_tables.R",
  "calibration_gap_reports.R",
  "confidence_bias_summary.R",
  "threshold_calibration_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R calibration workflows completed.")
